e2e/tests/lib/test_server_and_client_preparing/suite.go

```diff
@@ -0,0 +1,455 @@
+	s.runDaemonAtContainer(s.TestServerContainer, s.serverAccessdPath, "server_accessd.log")
+	time.Sleep(time.Second)
+	pidServerAccessd := s.firstProcessPIDAtContainer(s.TestServerContainer, s.serverAccessdPath)
+	Expect(pidServerAccessd).Should(BeNumerically(">", 0), "pid greater 0")
+
+	authKeysFilePath := filepath.Join("/home", cfg.User.Identifier, ".ssh", "authorized_keys")
+	contentAuthKeysFile := s.ExecuteCommandAtContainer(s.TestServerContainer,
+		[]string{"/bin/bash", "-c", "cat " + authKeysFilePath}, nil)
+	Expect(contentAuthKeysFile).To(HaveLen(1), "cat authorize should have one line text")
+	principal := calculatePrincipal(cfg.TestServer.ServerUUID, cfg.User.UUID)
+	Expect(contentAuthKeysFile[0]).To(MatchRegexp(".+cert-authority,principals=\""+principal+"\" ssh-rsa.{373}"),
+		"content should be specific")
+}
+
+//go:embed client_sock1.yaml
+var clientSocketCFG string
+
+//go:embed client_main.yaml
+var clientMainCFG string
+
+func (s *Suite) PrepareClientForSSHTesting(cfg flant_iam_preparing.CheckingSSHConnectionEnvironment) {
+	err := s.CheckFileExistAtContainer(s.TestClientContainer, s.authdPath, "f")
+	Expect(err).ToNot(HaveOccurred(), "TestClient should have authd")
+
+	err = s.CheckFileExistAtContainer(s.TestClientContainer, s.cliPath, "f")
+	Expect(err).ToNot(HaveOccurred(), "Test_сlient should have cli")
+
+	// Authd can be configured and run at Test_server
+	err = s.createIfNotExistsDirectoryAtContainer(s.TestClientContainer,
+		"/etc/flant/negentropy/authd-conf.d")
+	Expect(err).ToNot(HaveOccurred(), "folder should be created")
+
+	err = s.writeFileToContainer(s.TestClientContainer,
+		"/etc/flant/negentropy/authd-conf.d/main.yaml", clientMainCFG)
+	Expect(err).ToNot(HaveOccurred(), "file should be written")
+
+	err = s.writeFileToContainer(s.TestClientContainer,
+		"/etc/flant/negentropy/authd-conf.d/sock1.yaml", clientSocketCFG)
+	Expect(err).ToNot(HaveOccurred(), "file should be written")
+
+	err = s.writeFileToContainer(s.TestClientContainer,
+		"/opt/authd/client-jwt", cfg.UserJWToken)
+	Expect(err).ToNot(HaveOccurred(), "file should be written")
+
+	s.ExecuteCommandAtContainer(s.TestClientContainer,
+		[]string{"/bin/bash", "-c", "chmod 600 /opt/authd/client-jwt"}, nil)
+
+	s.killAllInstancesOfProcessAtContainer(s.TestClientContainer, s.authdPath)
+	s.runDaemonAtContainer(s.TestClientContainer, s.authdPath, "client_authd.log")
+	time.Sleep(time.Second)
+	pidAuthd := s.firstProcessPIDAtContainer(s.TestClientContainer, s.authdPath)
+	Expect(pidAuthd).Should(BeNumerically(">", 0), "pid greater 0")
+}
+
+func (s *Suite) getContainerByName(name string) (*types.Container, error) {
+	containers, err := s.dockerCli.ContainerList(context.Background(), types.ContainerListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range containers {
+		for _, n := range c.Names {
+			if n == "/"+name {
+				if c.State != "running" {
+					return nil, errors.New("Container with name " + name + " has state: " + c.State)
```

Comment:
Two cents: maybe fmt.Errorf would be more convenient

```go
fmt.Errorf("Container with name %s has state: %s", name, c.State)
```

Reply:
fixed